File: 1910.remove-all-occurrences-of-a-substring.py
#
# @lc app=leetcode id=1910 lang=python3
#
# [1910] Remove All Occurrences of a Substring
#

# @lc code=start
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        while part in s:
            i1 = s.find(part)
            i2 = i1 + len(part)
            s = s[:i1] + s[i2:]
        return s
# @lc code=end

File: js/1910.remove-all-occurrences-of-a-substring.js
/*
 * @lc app=leetcode id=1910 lang=javascript
 *
 * [1910] Remove All Occurrences of a Substring
 */

// @lc code=start
/**
 * @param {string} s
 * @param {string} part
 * @return {string}
 */
var removeOccurrences = function(s, part) {
  const i = s.search(part)
  if (i === -1)
    return s
  return removeOccurrences(s.slice(0,i) + s.slice(i+part.length), part)
};
// @lc code=end

